### ISSUE TITLE
Give the eraser a width

### ISSUE THREAD

**Issue by lovasoa** [lovasoa/whitebophir#8]
Currently, the eraser is just one pixel wide. It means that you have to be very precise when trying to erase a small shape, like a single point. We could make the eraser larger, so that you can just click less that X pixels away from a shape to erase it.

**Comment by adrian1001** [comment]
For long pencil strokes deleting the entire object is usually a bit too much.
Boils down to what is already mentioned in the eraser.js  `//TODO: add the ability to erase only some points in a line`.

Thoughts:
- Add function to boolean-subtract a shape from the drawing (there must be many libraries, e.g. https://github.com/w8r/GreinerHormann, or https://sourceforge.net/p/jsclipper/wiki/Home%206/)
- Add rectangle-eraser: click to mark corner of rectangle, drag to diagonal corner, upon release clip the drawing against rectangle
- Add a pen-like eraser:
  - option1: At every position, clip drawing against a circle-like polygon around cursor position
  - option2 (optimized): Tool draws a white shape to give user feedback; upon release, the shape is used to clip the drawing

Potential issues:
- performance: clipping a large drawing might require good algorithms (partitioning)
- cutting a line into segments will not look nice for thick lines because the end points have no information about at what angle the line was cut. (Draw line as polygon?)

**Comment by lovasoa** [comment]
Yes, these are good ideas. But all of them require a significant refactoring of the existing code. There is currently no spacial index of the shapes.
